### PR TITLE
Feat/true retention stats

### PR DESCRIPTION
**Pull request by L-M-Sherlock**
Related issue:
- #3147

Preview:

image

### CODE REVIEW

**Comment by dae** [comment]
- If this is not expected to change soon, perhaps we should be making it translatable from the start?
- The summary at the bottom is a bit long. Could we move it to the top, and make it simpler? E.g. 'Pass rate of cards with an interval >= 1 day'?

**Comment by dae** [comment]
Also, the green text doesn't fit so well with the rest of Anki's interface. Maybe standard black would be a better fit, with perhaps some bolding on key elements?

**Comment by L-M-Sherlock** [comment]
> * If this is not expected to change soon, perhaps we should be making it translatable from the start?

Good point! I will make it translatable later.

> * The summary at the bottom is a bit long. Could we move it to the top, and make it simpler? E.g. 'Pass rate of cards with an interval >= 1 day'?

The summary is written by @user1823 in https://github.com/open-spaced-repetition/fsrs4anki-helper/pull/200. I'd like to ask for his opinion.

> Also, the green text doesn't fit so well with the rest of Anki's interface. Maybe standard black would be a better fit, with perhaps some bolding on key elements?

OK.

**Comment by user1823** [comment]
> 'Pass rate of cards with an interval >= 1 day'?

Looks good.

**Comment by L-M-Sherlock** [comment]
image

**Comment by Expertium** [comment]
"7 days ago" and 30days ago" implies that it's retention for ONE specific day. "Last week" and Last month" is better. "Last year" would be nice, though this is already a ton of numbers.

**Comment by L-M-Sherlock** [comment]
OK

image

I also removed the Card Counts part because it's confusing and irrelevant to true retention.

**Comment by Expertium** [comment]
Will you add these as well?
image
I really hope so, I like knowledge acquisition rate, and I think a lot of people will like it too

File: ftl/core/statistics.ftl [hunk 1]
```diff
@@ -86,6 +86,18 @@ statistics-counts-learning-cards = Learning
 statistics-counts-relearning-cards = Relearning
 statistics-counts-title = Card Counts
 statistics-counts-separate-suspended-buried-cards = Separate suspended/buried cards
+statistics-true-retention-title = True Retention
+statistics-true-retention-subtitle = Pass rate of cards with an interval >= 1 day.
```
Comment (user1823) [inline ftl/core/statistics.ftl:90]:
```suggestion
statistics-true-retention-subtitle = Pass rate of cards with an interval ≥ 1 day.
```

**Comment by L-M-Sherlock** [comment]
> I really hope so, I like knowledge acquisition rate, and I think a lot of people will like it too

The stats has included some of them:

image

image

**Comment by Expertium** [comment]
Some, yes, but not all

**Comment by L-M-Sherlock** [comment]
Fine. I added the Estimated total knowledge (cards):

image

Daily load and retention by notes require more works to do, so I will consider it in another PR.

**Comment by Expertium** [comment]
I think these stats need their own dedicated section, with explanations copied from the add-on.

**Comment by dae** [comment]
Thanks Jarrett.

Some minor issues I noticed that I won't block on:

- collecting the results into a hashmap first is not the most efficient, but it doesn't really matter in this case
- total knowledge is probably more useful as an integer
- in some cases, the start of the table is cut off as the graphs window is resized

**Comment by dae** [comment]
(further changes welcome in follow-up PRs)

**Pull request merged**